### PR TITLE
Split operation with multiple return types

### PR DESCRIPTION
**Pull request by borsch**
This is a fix for issues like
- https://github.com/OpenAPITools/openapi-generator/issues/11731
- https://github.com/OpenAPITools/openapi-generator/issues/8701
- https://github.com/OpenAPITools/openapi-generator/issues/8700

There might be even more like those. The general idea of this PR is to split operation with different return types(content types) into multiple. While request for this feature was only for Spring, I did a global change since it was simpler to do in core, rather then in specific generator.

New flag is introduced to control this behavior and it's disabled by default

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Core team since this affects default generator: @wing328 @jimschubert @cbornet @jmini @etherealjoy @spacether

And java related committee since tests currently written for java/spring only:
@bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10)
@cachescrubber (2022/02) @welshm (2022/02) @MelleD (2022/02) @atextor (2022/02) @manedev79 (2022/02) @javisst (2022/02) @borsch (2022/02) @banlevente (2022/02)

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR. Please resolve the merge conflicts when you've time.

File: modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/OpenApiGeneratorPlugin.kt [hunk 1]
```diff
@@ -138,6 +138,7 @@ class OpenApiGeneratorPlugin : Plugin<Project> {
                     generateModelDocumentation.set(generate.generateModelDocumentation)
                     generateApiTests.set(generate.generateApiTests)
                     generateApiDocumentation.set(generate.generateApiDocumentation)
+                    splitOperation.set(generate.splitOperation)
```
Comment (wing328) [inline modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/OpenApiGeneratorPlugin.kt:141]: Do we need to make similar change in the CLI as well?

Comment (borsch) [reply modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/OpenApiGeneratorPlugin.kt:141]: @wing328 you talk about this repo https://github.com/OpenAPITools/openapi-generator-cli ?

Comment (borsch) [reply modules/openapi-generator-gradle-plugin/src/main/kotlin/org/openapitools/generator/gradle/plugin/OpenApiGeneratorPlugin.kt:141]: will check

**Comment by borsch** [comment]
@spacether my implementation is cross language/library

**Comment by cv** [comment]
Hi folks! Currently bumping into this issue in the TypeScript fetch generator, so having a global way to deal with it seems like a great way to go.

**Comment by Christina0114** [comment]
Hi, I try it on my own open api file, i find the names of generated functions are not very good. Do you ever consider multiple request content types?

**Comment by borsch** [comment]
@Christina0114 what do you mean by "not very good names"?

> Do you ever consider multiple request content types?
Not as for now

**Comment by borsch** [comment]
Hi @wing328 
Could you please review this PR?

**Comment by spacether** [comment]
> @spacether my implementation is cross language/library

Using the Java layer data is also a is cross language/library solution
```
Our code already store multiple response media types in the Java layer in CodegenResponse.content
Have you considered using this information to write one function per content type at the template level?
```
In 3.0.X specs, a response doe not have a datatype. It's content object has a datatype per the spec:
https://github.com/OAI/OpenAPI-Specification/blob/main/versions/3.0.3.md#response-object
Continuing to store and use data type at the CodegenResponse level makes moving to openapi v3.1.0 spec support much more difficult.

Please consider using [CodegenResponse.content](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenResponse.java#L91) as it has the data that you need
You can iterate over it and generate multiple interfaces/fns/classes with separate return types.
That is exactly what [this code does here](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/resources/python-experimental/endpoint.handlebars#L290).

**Pull request closed without merging**